robot.py

```diff
@@ -45,31 +47,39 @@ def robotInit(self):
         # Create a simple timer (docs: https://robotpy.readthedocs.io/projects/wpilib/en/latest/wpilib/Timer.html#wpilib.timer.Timer.get)
         self.timer = wpilib.Timer()
         
+        self._state = State(self, FREEZE)
+        
+    @property
+    def state(self):
+        return self._state.state
+        
+    @state.setter
+    def state(self, new_state):
+        self._state.dispatch(new_state)
+        
     def autonomousInit(self):
         """Called only at the beginning of autonomous mode."""
         self.timer.reset()
         self.timer.start()
 
     def autonomousPeriodic(self):
         """Called every 20ms in autonomous mode."""
-        
-        # Drive forward for two seconds.
-        if self.timer.get() < 2:
-          self.myRobot.tankDrive(1, 1)
+        self._state.update()
```

Comment:
Might want to change the `state` class to `states` or something to avoid confusion with `_` characters.